### PR TITLE
template_name none if fields empty #7973

### LINKED ISSUE #7973: consider migration for old infowindow templates
when working on infowindows: type in tabmenu (https://github.com/CartoDB/cartodb/pull/7972) I spotted when selecting an old infowindow (i.e. "infowindow_light_header_orange") in old editor will make new builder crash

as a temporal mitigation for this I added the templates but I think it's time to consider a migration for several reasons, mostly:

- as old templates won't exist in the carousel (see attached) it wouldn't appear selected, when color template should appear selected
- for consistency reasons, color template is still named "infowindow_light_header_blue", that's definitely not the best name

I'd propose set the template html for the `infowindow_light_header_green, infowindow_light_header_orange, infowindow_light_header_yellow` occurences, so it would appear as custom template

and rename `infowindow_light_header_blue` to `infowindow_color`

finally remove the legacy templates

/cc @juanignaciosl @xavijam

### PR DESCRIPTION
**Pull request by juanignaciosl**
This closes #7973.

### CODE REVIEW

**Comment by juanignaciosl** [comment]
@javitonino CR this, please.

cc @matallo @xavijam

**Comment by javitonino** [comment]
Please, ping me again when tests pass (seems like an import error).

**Comment by juanignaciosl** [comment]
@javitonino fixed, CR, please.

File: spec/requests/carto/api/infowindow_migrator_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      "alternative_names" => {},
+      "maxHeight" => 180
+    }
+  end
+
+  describe '#migrate_builder_infowindow' do
+    it 'sets template_name "none" if fields are empty' do
+      migrated = migrator.migrate_builder_infowindow(infowindow)
+      migrated['template_name'].should eq 'none'
+      migrated['template'].should eq ''
```
Comment (javitonino) [inline spec/requests/carto/api/infowindow_migrator_spec.rb]: I think this check does not hold true if the source infowindow has `template` set. I mean, the migrator doesn't seem to clear this attribute when setting `template_name = none`. If this is important (I am guessing it is, as you are explicitly testing it here), maybe the test should check that it is being cleared, by adding something to template in the input.

If this is not important, disregard.

**Comment by javitonino** [comment]
Left a comment, not sure if it is important (it depends on expected output). Take a look, and :+1:

File: spec/requests/carto/api/infowindow_migrator_spec.rb (hunk 1 above)
Comment (juanignaciosl) [inline spec/requests/carto/api/infowindow_migrator_spec.rb]: Yes, I know. I added that in order to make sure that `template` wasn't added, which was the previous behaviour :-)

Comment (javitonino) [inline spec/requests/carto/api/infowindow_migrator_spec.rb]: :sparkles: then

**Pull request merged**